src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -219,28 +219,28 @@ public SignatureTrustAndValidityVerificationProvider(IEnumerable<KeyValuePair<st
             Timestamp timestamp;
             var timestampSummary = GetTimestamp(signature, verifierSettings, out timestamp);
 
+            var status = signature.Verify(
+            timestamp,
+            settings,
+            _fingerprintAlgorithm,
+            certificateExtraStore);
+
             if (timestampSummary.Status != SignatureVerificationStatus.Valid && !verifierSettings.AllowIgnoreTimestamp)
             {
                 return new SignatureVerificationSummary(
                     signature.Type,
                     SignatureVerificationStatus.Disallowed,
                     SignatureVerificationStatusFlags.NoValidTimestamp,
-                    timestampSummary.Issues);
+                    status.Issues.Concat(timestampSummary.Issues));
             }
 
-            var status = signature.Verify(
-                timestamp,
-                settings,
-                _fingerprintAlgorithm,
-                certificateExtraStore);
-
             return new SignatureVerificationSummary(
                 status.SignatureType,
                 status.Status,
                 status.Flags,
                 status.Timestamp,
                 status.ExpirationTime,
-                timestampSummary.Issues.Concat(status.Issues));
+                status.Issues.Concat(timestampSummary.Issues));
```

Comment:
May I know if those changes are for fixing https://github.com/NuGet/Home/issues/10535?
I understand that we don't have test infrastructure for untrustedroot for timestamp, so the test is not added.
May I know if you verified in other ways (e.g. manually patch and test) to make sure https://github.com/NuGet/Home/issues/10535 is fixed? Thanks.

Reply:
> May I know if those changes are for fixing NuGet/Home#10535?

Yes, that is correct.  I verified this fix by modifying the asserts statements in the following tests.
     
- `SignatureTrustAndValidityVerificationProviderTests.GetTrustResultAsync_WithNoIgnoringTimestamp_TimestampWithGeneralizedTimeOutsideCertificateValidity_FailAsync()` https://github.com/NuGet/NuGet.Client/pull/3992/files#diff-ac0c57f69076c8fd3f9e38cdbcca84fa2de8e9052cbe91d9fac7e108f57d2b84R556
- `PackageSignatureVerifierTests.VerifySignaturesAsync_WithTimestampChainingToUntrustedRoot_NotAllowIgnoreTimestamp_FailAsync()` https://github.com/NuGet/NuGet.Client/pull/3992/files#diff-6a15bb307e86dc7012938479f8a3fe9aff77788c02854afca190d0a2e5b4e405R768

I have also verified this fix manually patching dotnet SDK.

Reply:
That's great! 
Thanks for correcting me.
Just found the setting `AllowIgnoreTimestamp` for VerifyCommandDefaultPolicy is set to false.
And the two test cases set `AllowIgnoreTimestamp` to false.
So the two test cases could repro the same issue with NuGet/Home#10535.